pkg/util/azureerrors/error.go

```diff
@@ -149,3 +149,17 @@ func ResourceGroupNotFound(err error) bool {
 	}
 	return false
 }
+
+func IsInvalidIdentityUserError(err error) bool {
+	var detailedErr autorest.DetailedError
```

Comment:
Let's use `azcore` instead of autorest.  Then you won't have to cast the int below.  

`azcore.ResponseError`

Reply:
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore#ResponseError